src/plugins/data/common/exports/export_csv.tsx

```diff
@@ -22,39 +20,27 @@ interface CSVOptions {
   escapeFormulaValues: boolean;
   formatFactory: FormatFactory;
   raw?: boolean;
-  columnsSorting?: string[];
 }
 
 export function datatableToCSV(
   { columns, rows }: Datatable,
-  { csvSeparator, quoteValues, formatFactory, raw, escapeFormulaValues, columnsSorting }: CSVOptions
+  { csvSeparator, quoteValues, formatFactory, raw, escapeFormulaValues }: CSVOptions
 ) {
   const escapeValues = createEscapeValue({
     separator: csvSeparator,
     quoteValues,
     escapeFormulaValues,
   });
 
-  const sortedIds = columnsSorting || columns.map((col) => col.id);
-
-  // Build an index lookup table
-  const columnIndexLookup = sortedIds.reduce((memo, id, index) => {
-    memo[id] = index;
-    return memo;
-  }, {} as Record<string, number>);
-
-  // Build the header row by its names
   const header: string[] = [];
   const sortedColumnIds: string[] = [];
   const formatters: Record<string, ReturnType<FormatFactory>> = {};
```

Comment:
@dej611 I removed the column sort logic from here altogether. I was trying to find the best approach based on what we talked about and I think instead of having the vis provide the sort order of the columns we just have the vis provide the data to be exported. 

This was the columns and rows are sorted as needed in the new `vis.getExportDatatables` method.

Additionally, the `downloadCsvShareProvider.getShareMenuItems` loads all tables from the `activeData`, so when there are multiple it is confusing to have a single list of sorted columns. Now if a vis needs to export more than one datatable, it can sort/filter any active datatables before passing them to the exporter.

LMK what you think...